test/test_selectors.py

```diff
@@ -0,0 +1,670 @@
+        self.assertEqual(rd.recv(1), b'x')
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_multiple_interrupts_with_event(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+        key = s.get_key(rd)
+
+        def second_alarm(*args):
+            wr.send(b'x')
+
+        def first_alarm(*args):
+            self.set_alarm(SHORT_SELECT)
+            signal.signal(signal.SIGALRM, second_alarm)
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, first_alarm)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+        self.addCleanup(signal.alarm, 0)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        t = get_time()
+        self.assertEqual([(key, selectors.EVENT_READ)], s.select(LONG_SELECT))
+        t = get_time() - t
+        self.assertGreaterEqual(t, SHORT_SELECT * 2)
+        self.assertLess(t, LONG_SELECT)
+        self.assertEqual(rd.recv(1), b'x')
+
+    # Test ensures that _syscall_wrapper properly raises the
+    # exception that is raised from an interrupt handler.
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_interrupt_exception(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        class AlarmInterrupt(Exception):
+            pass
+
+        def alarm_exception(*args):
+            raise AlarmInterrupt()
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, alarm_exception)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        t = get_time()
+        self.assertRaises(AlarmInterrupt, s.select, LONG_SELECT)
+        self.assertLess(get_time() - t, LONG_SELECT)
+
+    def test_fileno(self):
+        s = self.make_selector()
+        if hasattr(s, "fileno"):
+            fd = s.fileno()
+            self.assertTrue(isinstance(fd, int))
+            self.assertGreaterEqual(fd, 0)
+
+    # According to the psutil docs, open_files() has strange behavior
+    # on Windows including giving back incorrect results so to
+    # stop random failures from occurring we're skipping on Windows.
+    @skipIf(sys.platform == "win32", "psutil.Process.open_files() is unstable on Windows.")
+    def test_leaking_fds(self):
+        proc = psutil.Process()
+        before_fds = len(proc.open_files())
+        s = self.make_selector()
+        s.close()
+        after_fds = len(proc.open_files())
+        self.assertEqual(before_fds, after_fds)
+
+
+def min_mac_version(major, minor):
+    if sys.platform == "darwin":
+        try:
+            if tuple(map(int, platform.mac_ver()[0].split("."))) < (major, minor):
+                return False
+        except ValueError:
+            pass
```

Comment:
I actually don't know, the asyncio/test_selectors has this test and doesn't say anything about why the ValueError is there. Should I remove it and see what happens?